basex/src/main/java/com/formulasearchengine/mathosphere/basex/TexQueryGenerator.java

```diff
@@ -11,81 +18,102 @@
+        ArrayList<NameValuePair> pDefault = new ArrayList<>();
+        pDefault.add(new BasicNameValuePair("format", "xhtml"));
+        pDefault.add(new BasicNameValuePair("whatsin", "math"));
+        pDefault.add(new BasicNameValuePair("whatsout", "math"));
+        pDefault.add(new BasicNameValuePair("cmml", ""));
+        pDefault.add(new BasicNameValuePair("nodefaultresources", ""));
+        pDefault.add(new BasicNameValuePair("preload", "LaTeX.pool"));
+        pDefault.add(new BasicNameValuePair("preload", "article.cls"));
+        pDefault.add(new BasicNameValuePair("preload", "amsmath.sty"));
+        pDefault.add(new BasicNameValuePair("preload", "amsthm.sty"));
+        pDefault.add(new BasicNameValuePair("preload", "amstext.sty"));
+        pDefault.add(new BasicNameValuePair("preload", "amssymb.sty"));
+        pDefault.add(new BasicNameValuePair("preload", "eucal.sty"));
+        pDefault.add(new BasicNameValuePair("preload", "[dvipsnames]xcolor.sty"));
+        pDefault.add(new BasicNameValuePair("preload", "url.sty"));
+        pDefault.add(new BasicNameValuePair("preload", "hyperref.sty"));
+        pDefault.add(new BasicNameValuePair("preload", "mws.sty"));
+        pDefault.add(new BasicNameValuePair("preload", "texvc"));
+        return pDefault;
+    }
+
+    void setParams(List<NameValuePair> params) {
+        this.params = params;
+        lateXMLConfig.setParams(updateParamFormat(params));
+    }
+
+    private Map<String, Object> updateParamFormat(List<NameValuePair> params) {
+        HashMap<String, Object> map = new HashMap<>();
+        for (NameValuePair pair : params) {
+            if (map.containsKey(pair.getName())) {
+                Object oSetting = map.get(pair.getName());
+                if (oSetting instanceof List) {
+                    ((List<String>) oSetting).add(pair.getValue());
+                } else {
+                    assert (oSetting instanceof String);
+                    map.put(pair.getName(), Lists.newArrayList(pair.getValue(), oSetting));
+                }
+            } else {
+                map.put(pair.getName(), pair.getValue());
+            }
+        }
+        return map;
+    }
+
+    String getLaTeXMLURL() {
+        return lateXMLConfig.getUrl();
+    }
+
+    Map getOb() {
+        HashMap<String, Object> hashMap = new HashMap();
+        hashMap.put("status_code", serviceResponse.getStatusCode());
+        hashMap.put("status", serviceResponse.getStatus());
+        hashMap.put("log", serviceResponse.getLog());
+        hashMap.put("result", serviceResponse.getResult());
+        return hashMap;
+    }
+
+    void setLaTeXMLURL(String laTeXMLURL) {
+        lateXMLConfig.setUrl(laTeXMLURL);
+    }
+
+    String request(String tex) throws IOException, IllegalStateException {
+        LaTeXMLConverter converter = new LaTeXMLConverter(lateXMLConfig);
+        try {
+            serviceResponse = converter.convertLatexmlService(tex);
```

Comment:
I had to look at the try-catch blocks of https://github.com/ag-gipp/MathMLConverters/blob/master/src/main/java/com/formulasearchengine/mathmlconverters/latexml/LaTeXMLConverter.java#L94-L113 to understand how the errors are handled. To adjust my test cases.